source/Engine.cpp

```diff
@@ -837,8 +837,19 @@ void Engine::Step(bool isActive)
 		targetSwizzle = target->GetSwizzle();
 		info.SetString("mission target", target->GetPersonality().IsTarget() ? "(mission target)" : "");
 
+		// Only update the "active" state shown for the target if it is
+		// in the current system and targetable, or owned by the player.
 		int targetType = RadarType(*target, step);
-		info.SetOutlineColor(GetTargetOutlineColor(targetType));
+		const bool blinking = targetType == Radar::BLINK;
+		if(!blinking && ((target->GetSystem() == player.GetSystem() && target->IsTargetable()) || target->IsYours()))
+			targetWasInactive = targetType == Radar::INACTIVE;
+		if(blinking)
+			info.SetOutlineColor(GetTargetOutlineColor(Radar::BLINK));
+		else if(targetWasInactive)
+			info.SetOutlineColor(GetTargetOutlineColor(Radar::INACTIVE));
+		else
+			info.SetOutlineColor(GetTargetOutlineColor(targetType));
```

Comment:
Another option might be to handle blinking first.

```suggestion
		if(targetType == Radar::BLINK || target->IsYours())
			info.SetOutlineColor(GetTargetOutlineColor(targetType));
		else
		{
			if(target->GetSystem() == player.GetSystem() && target->IsTargetable())
				lastTargetType = targetType;
			info.SetOutlineColor(GetTargetOutlineColor(lastTargetType));
		}
```

Reply:
(I'm not sure why GitHub decided to make this a separate comments-thread.)